kratos/includes/registry_item.h

```diff
@@ -66,6 +67,131 @@ class RegistryItem
     /// Pointer definition of RegistryItem
     KRATOS_CLASS_POINTER_DEFINITION(RegistryItem);
 
+    /// Subregistry item type definition
+    using SubRegistryItemType = std::unordered_map<std::string, Kratos::shared_ptr<RegistryItem>>;
+
+    /// Custom iterator with key as return type to be used in the Python export
+    class key_return_iterator
+    {
+    public:
+        ///@name Type Definitions
+        ///@{
+
+        using BaseIterator      = SubRegistryItemType::iterator;
```

Comment:
As far as I can tell, ```key_return_iterator``` is not supposed to mutate the underlying keys, so ```BaseIterator``` should reflect that:
```suggestion
        using BaseIterator      = SubRegistryItemType::const_iterator;
```

Reply:
Agree and done. Thanks for the observation.